src/DI/Container.php

```diff
@@ -108,4 +111,17 @@ public function tabs() {
 	public function ui() {
 		return new UIServices($this);
 	}
+
+    /**
+     * Get interface to the http service.
+     *
+     * @return GlobalHttpState
+     */
+	public function http() {
+	    return new HTTPServices(
```

Comment:
Hi @d3r1w,

this defeats the purpose of the dependency injection container. No dependency injected here, static binding to the two implementations instead. Do you understand the purpose and mechanism of DI?

The `UIServices` from [l.109](https://github.com/ILIAS-eLearning/ILIAS/pull/466/files#diff-f46de88208848386905f0e063e513b62L109) is just for structuring the access to the DIC-properties further and does not contain the real UI-services on its own. These are added to the container in ilInitialisation, which would imo would be the proper place to add the HTTPServices and the two required implementations as well.

I'm not convinced we should expose the GlobalHTTPState via a method on the Container. Why not inject it in the services that requires the state as well?

Best regards!